modules/library/main/src/main/java/org/geotools/styling/Logarithmic.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.styling;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.opengis.filter.expression.Expression;
+import org.opengis.style.ContrastMethod;
+
+/**
+ * @author ian
+ *
+ */
+public class Logarithmic extends AbstractContrastEnhancementMethod {
+
+    final static List<String> PARAM_NAMES = Arrays.asList("normalizationFactor", "correctionFactor");
```

Comment:
This is really tough, if we look at RasterSymbolizer it lists the keys as public string constants so we can have appropriate javadocs to reference.

In this case you may wish to add the keys to ContrastEnhancement and static include them here.

````
final static List<String> PARAM_NAMES = Arrays.asList( NORMALIZATION_FACTOR, CORRECTION_FACTOR);
````